src/routing.ts

```diff
@@ -348,7 +352,8 @@ export function createRouterContext(
         setSource({
           value: next,
           replace: first.replace,
-          scroll: first.scroll
+          scroll: first.scroll,
+          state: first.state
```

Comment:
I'm not sure this is exactly what we want here, but maybe it is. The router batches synchronous navigations - that is to say if you navigate with a link or `useNavigate`, if another navigation occurs during rendering from something like a `<Navigate />` component we treat them as one navigation. So here `first` refers to the original navigation event which would mean if a redirect provided a state as well, that would be lost. The alternatives are pass in the state to this function and always use it or maybe use the last one defined.

Reply:
If we will use `useNavigate` with custom state, and this redirects with `<Navigate />`, we will lost the state and got undefined. `react-router` and native `.pushState` works in the same way. So, I suppose it works as expected right now, or I messed something.

Reply:
I agree, if a redirect occurs, then the state should be whatever was used on that redirect navigation which likely will be undefined. The only problem as it stands is there is a mismatch between what gets set in the `state` signal and what gets passed to the integration. The state signal will get the state from the last navigation event and the integration will be sent the first event's state.

Here is an example: https://codesandbox.io/s/solid-app-router-pr53-1-0c3ek
- Click `foo` link, notice state is `{ from: 'home' }
- Click `baz` link, notice state is `undefined`
- Click browser back, notice state is `{ from: 'home' }` (expected)
- Click browser forward, notice state is `{ from: 'foo' }` (unexpected, should be `undefined`)

Reply:
My latest commits makes this work like I would expect. Example (use the same steps as above): https://codesandbox.io/s/solid-app-router-pr53-2-un0bi